native/src/modules/navigation/InternalPathnameParser.js

```diff
@@ -0,0 +1,199 @@
+// @flow
+
+import {
+  CATEGORIES_ROUTE,
+  DASHBOARD_ROUTE,
+  DISCLAIMER_ROUTE,
+  EVENTS_ROUTE,
+  LANDING_ROUTE,
+  LOCAL_NEWS_TYPE,
+  NEWS_ROUTE,
+  OFFERS_ROUTE,
+  POIS_ROUTE,
+  SEARCH_ROUTE,
+  SPRUNGBRETT_OFFER_ROUTE,
+  TU_NEWS_TYPE
+} from 'api-client/src/routes'
+import type {
+  LocalNewsType,
+  TuNewsType
+} from 'api-client/src/routes'
+import type { RouteInformationType } from './createNavigate'
+
+const FEATURE_ID_INDEX = 3
+
+class InternalPathnameParser {
+  _pathname: string
+  _parts: Array<string>
+  _length: number
+  _languageCode: string
+  _fixedCity: string | null
+
+  constructor (pathname: string, languageCode: string, fixedCity: string | null) {
+    this._pathname = pathname
+    this._fixedCity = fixedCity
+    this._parts = this.pathnameParts(pathname)
+    this._length = this._parts.length
+    this._languageCode = languageCode
+  }
+
+  pathnameParts = (pathname: string) => {
+    return pathname.split('/').filter(Boolean)
+  }
+
+  isFixedCity = () => {
```

Comment:
Why is this called fixedCity and not just city? Should this be `isCityFixed`?

Reply:
Because this is just used to check if the city is the same as the fixed city from the build config (i.e. the one passed in the constructor). I think the terminology is right :see_no_evil:

Reply:
Ah I did not get that fixedCity is the naming from the buildConfig